filip/semantics/semantic_models.py

```diff
@@ -0,0 +1,1683 @@
+
+    def get_info(self) -> str:
+        """
+        Retrieve the executed command result from the IoT-Device
+
+        Raises:
+            Exception: If the command was not yet saved to Fiware
+        """
+        return self._get_field_from_fiware(field_name=f'{self.name}_info',
+                                           required_type="commandResult").value
+
+    def get_status(self):
+        """
+        Retrieve the executed command status from the IoT-Device
+
+        Raises:
+            Exception: If the command was not yet saved to Fiware
+        """
+        return self._get_field_from_fiware(field_name=f'{self.name}_status',
+                                           required_type="commandStatus").value
+
+    def get_all_field_names(self) -> List[str]:
+        """
+        Get all the field names that this command will add to Fiware
+        """
+        return [self.name, f"{self.name}_info", f"{self.name}_result"]
+
+    class Config:
+        """if the name is changed the attribute needs to be removed
+        and re-added to the device. With frozen that logic is more clearly
+        given in the library. Further it allows us to hash the object"""
+        frozen = True
+
+
+class DeviceAttributeType(str, Enum):
+    """
+    Retrieval type of the DeviceAttribute value from the IoT Device into Fiware
+    """
+    lazy = "lazy"
+    """The value is only read out if it is requested"""
+    active = "active"
+    """The value is kept up-to-date"""
+
+
+class DeviceAttribute(DeviceProperty):
+    """
+    Model describing an attribute property of an IoT device.
+
+    The attribute will add one field to the fiware instance:
+        - {NameOfInstanceField}_{Name}, holds the value of the Iot device
+        attribute: get_value()
+
+    A DeviceAttribute can only belong to one field of one instance. Assigning
+    it to multiple fields will result in an error.
+    """
+    attribute_type: DeviceAttributeType
+    """States if the attribute is read actively or lazy from the IoT Device 
+    into Fiware"""
+
+    def get_value(self):
+        """
+        Retrieve the current value from the Iot Device
+
+        Raises:
+            Exception: If the DeviceAttribute was not yet saved to Fiware
+        """
+        return self._get_field_from_fiware(
+            field_name=f'{self._instance_link.field_name}_{self.name}',
+            required_type="StructuredValue").value
+
+    def get_all_field_names(self) -> List[str]:
+        """
+        Get all the field names that this command will add to Fiware
+        """
+        return [f'{self._instance_link.field_name}_{self.name}']
+
+    class Config:
+        """if the name or type is changed the attribute needs to be removed
+        and readded to the device. With frozen that logic is more clearly
+        given in the library. Further it allows us to hash the object"""
+        frozen = True
+        use_enum_values = True
+
+
+class Field(BaseModel):
```

Comment:
this might conflict with pydantic Field

Reply:
Yes, but only at one point i needed to use pydantic.Fiedl ot avoid conflict. As normaly this Field is not imported in other modules. And the name describes the best what it is